agent/consul/discoverychain/compile.go

```diff
@@ -128,9 +129,22 @@ func (c *compiler) recordServiceProtocol(serviceName string) error {
 	if serviceDefault := c.entries.GetService(serviceName); serviceDefault != nil {
 		return c.recordProtocol(serviceName, serviceDefault.Protocol)
 	}
+	if c.entries.GlobalProxy != nil {
```

Comment:
Isn't this guaranteed to cause an error if the protocol is set globally and for a service?

Reply:
Basically if we call `recordProtocol` with the `service-defaults` protocol and follow it up with another `recordProtocol` with the proxy-defaults protocol and they are not the same thing, then I believe you are going to throw a ConfigEntryGraphError

Reply:
If it is set for the service it enters the first conditional and returns, rather than hitting this second conditional.

Reply:
Yep, sorry. I missed the whole "return" bit in there.